src/misc/unicode.cpp

```diff
@@ -1284,6 +1679,137 @@ static void construct_decomposed(const code_page_mapping_t& normalized,
 	}
 }
 
+static void construct_box_fallback(const map_grapheme_to_dos_t& code_page_mapping,
+                                   map_box_code_points_t& out_box_code_points)
+{
+	auto is_adaptation_needed = [&]() {
+		// Check if current list of box drawing characters needs
+		// to be adapted due to missing characters in the given
+		// code page
+		for (const auto& [from, target] : out_box_code_points) {
+			if (!code_page_mapping.count(target)) {
+				return true;
+			}
+		}
+		return false;
+	};
+
+	auto is_alias_group_suitable = [&](const std::vector<alias_t>& alias_group) {
+		// Check if the given alias group is:
+		// - applicable = the code page contains all the characters
+		//   corresponding to target code points
+		// - profitable = applying the alias will reduce the number
+		//   of code points without matching characters in the given
+		//   code page
+
+		bool is_group_profitable = false;
+		for (const auto& [from, target] : out_box_code_points) {
+			for (const auto& alias : alias_group) {
+				if (from != alias.first) {
+					continue;
+				}
+
+				if (!code_page_mapping.count(alias.second)) {
+					// We cannot apply this alias group
+					return false;
+				}
+
+				if (!code_page_mapping.count(target)) {
```

Comment:
Suggest quantity comparisons (count == 0)

Reply:
OK, changed - everywhere in this file.